cherrypy/lib/static.py

```diff
@@ -75,6 +75,13 @@ def serve_file(path, content_type=None, disposition=None, name=None,
         i = path.rfind('.')
         if i != -1:
             ext = path[i:].lower()
+            if ext in mimetypes.encodings_map.keys():
```

Comment:
Well, looks good, but too abstract. What is the `real filename extension`? How is the encoding determined? What will happen if I call the file `test.js.42.gz`, for example? Add tests for corner cases please.